src/rev_ai/apiclient.py

```diff
@@ -1,34 +1,27 @@
 # -*- coding: utf-8 -*-
 """Speech recognition tools for using Rev.ai"""
 
-import requests
 import json
-from requests.exceptions import HTTPError
 from .models import Job, Account, Transcript, CaptionType
-from . import __version__
+from .baseclient import BaseClient
 
 try:
     from urllib.parse import urljoin
 except ImportError:
     from urlparse import urljoin
 
 
-class RevAiAPIClient:
+class RevAiAPIClient(BaseClient):
```

Comment:
I wonder if we can simplify the tests now that we have this separation. In theory we should be able to mock out the `_make_http_request` call and reduce a lot of duplicated checks in the tests 

@k-weng how easy do you think it would be to do this

Reply:
I'm actually currently modifying the tests for `_make_http_request`. Removed from `test_apiclient` and moved it to `test_baseclient`. It makes sense to me and seems to work just fine. Is that what you mean @kbridbur ?

Reply:
@KostasRev, what @kbridbur is referring to is the test pattern we use in the revai node sdk. One of the benefits of extracting the baseclient is that we can independently test the baseclient and remove any excessive test assertions for asserting calls to the httpclient directly.

let's zoom tomorrow if we need to discuss it more.